AdminAction.ts

```diff
@@ -0,0 +1,17 @@
+import { Field, Struct } from "o1js"
+
+/** Enumerates the different types of administrative actions that can be performed on a token. */
+export class AdminAction extends Struct({
+  type: Field,
+}) {
+  public static types = {
+    mint: 0,
+    setTotalSupply: 1,
+    setPaused: 2,
+    setAdmin: 3,
```

Comment:
I think the `AdminAction` needs to include not only the type, but also the full info about the action that is being done, e.g.
* `mint`: the amount and receiver, or even the minting account update
* `setAdmin`: new public key of the admin contract
etc

If we don't include this info, it's not feasible to write a smart contract that approves admin actions without a signature

Reply:
Good point! I've now spread the `canAdmin` method into dedicated methods for different actions, and each get appropriate information on what is actually happening.